### DISCUSSION TITLE
The map/cursor does not rotate

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#303] [Q&A] [answered]
Such a situation: driving slowly ~10 km/h, the map rotation was blocked. It's probably due to `Settings | Location | Advanced | Orientation threshold -> 20 km/h` (for car/motorcycle is the default value?) But perhaps the cursor should rotate in the direction of travel? 
In this case, I was moving as shown by the arrow, and the cursor did not rotate. He was moving to the left, in a vertical position. Is this expected behavior under these conditions?

<img src="https://github.com/devemux86/cruiser/assets/122647782/3637f6e9-6a48-4701-827a-b37e37ccf198" width="50%"/>

**Comment by devemux86** [comment]
@Ciutnik thanks for the report.

> It's probably due to Settings | Location | Advanced | Orientation threshold -> 20 km/h (for car/motorcycle is the default value?)

There is an adjustable threshold (as speed) option, above which the GPS heading is considered.
Now it is different per profile, default for driving 20km/h, cycling 5km/h, walking 2km/h.

e.g. avoid wrong map rotation when we stop and GPS sensor sends random headings.

Did you drive, with what speed?

**Reply by devemux86** [reply]
But it should use the last heading, not reset to zero (like in image?).

@Ciutnik is this something new?
Doesn't the direction works correctly all this time?

**Reply by Ciutnik** [reply]
I was driving at a speed certainly below the orientation threshold. I think about 10 km/h. 

I'm not sure if the cursor orientation **reset**. Perhaps the blocked direction is in line with the direction ahead of the first left turn. So if that was the case, this behavior is expected... So shouldn't the arrow be a big dot?

The first time I noticed it, I immediately took a screenshot.

**Reply by Ciutnik** [reply]
> But perhaps the cursor should rotate in the direction of travel?

I just realized I'm talking nonsense 😀 After all, that's what the orientation threshold is for, to block it.

**Reply by 0709wiwiwi** [reply]
> So shouldn't the arrow be a big dot?

I would prefer to observe a slowly moving big dot IF the speed is indeed under the treshold to replace that now so comic non rotating strange arrow cursor.



.

**Reply by devemux86** [reply]
If the GPS reports that there is _not available_ location heading then the marker is the dot.
If the GPS reports _an available_ location heading then the marker is the arrow (in navigation / follow).

With the orientation threshold, we simply select the limit above which the reported heading is considered,
or the last valid heading is used.

**Reply by Ciutnik** [reply]
OK. I see. Fortunately, the strange behavior of the cursor occurs very rarely and only under certain conditions, so we can live with it. Especially since lowering the rotation threshold should help.

**Comment by devemux86** [comment]
@Ciutnik you can lower the orientation threshold < 20 km/h.
Should we have a lower default value for driving?

Anyway I implemented an improvement in case the GPS temporarily can't report the speed.

**Reply by Ciutnik** [reply]
Thanks! I lowered the threshold to 10 km/h immediately after this event 🙂 Maybe it's a better value for a motorcycle, because the device is in the open air and possible GPS errors are smaller. In the car, under the roof, when GPS errors can cause the location to float, 20 km/h will be fine.

**Reply by devemux86** [reply]
@Ciutnik indeed we can lower the defaults to avoid such cases.

**Comment by devemux86** [comment]
General note:

Some apps don't respect the GPS direction while driving.
They snap on road direction that can have wrong results.

Then they have many errors, e.g. on turns where they incorrectly snap the location
on wrong roads and the arrow / map is slow and points to wrong random directions.

**Comment by 0709wiwiwi** [comment]
Yep Emux had already understood that before. It only concerns what I think is a rather strange current presentation anyway. I (and possibly others) thought when I first saw this to occur that it was a "defect" within the Cruiser app.
I then realised afterwards that this was a presentation choice in Cruiser. However, I still do find it an odd choice. As the Cruiser app attaches great importance into a most nice presentation, it is so still strange. So I just share my preference, which I think in this case is a slightly neater presentation once you get below that threshold. Anyway a position dot (=circle) does not point into a false "random" direction ;-)

**Comment by devemux86** [comment, marked as answer]
> we can lower the defaults to avoid such cases.

Implemented in [Cruiser 3.0.32](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
